### PR TITLE
add read only mode of drag-data-store, fixes #95

### LINKED ISSUE #95: Declare that `dataTransfer` should be read-only unless created by JS
Currently, there is no detail of `dataTransfer` attribute value of `input` nor `beforeinput` event.

At least I've tested, Chrome sets `dataTransfer` to read-only DataTransfer instance. For example, `clearData()` does not clear its data. I think that this is reasonable behavior because if we need to allow to modify the DataTransfer instance, editor of browser needs to copy the instance every time. This causes editor performance too bad since memory allocation is too slow but source of DataTransfer may be too big.

On the other hand, if the InputEvent instance is created by JS, whether the DataTransfer object is read-only, read-write or protected should depend on the DataTransfer instance which is set via the dictionary at constructing. I.e., when InputEvent is created with a DataTransfer instance, InputEvent should set its `dataTransfer` to the instance itself. Then, InputEvent does not need to clone given DataTransfer instance.

### PR DESCRIPTION
**Pull request by johanneswilm**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/input-events/pull/99.html" title="Last updated on Jun 30, 2019, 6:38 AM UTC (2687f75)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/input-events/99/184ea1a...2687f75.html" title="Last updated on Jun 30, 2019, 6:38 AM UTC (2687f75)">Diff</a>

### CODE REVIEW

**Comment by siusin** [comment]
Hi @johanneswilm .

Per the [MoU between W3C and WHATWG](https://www.w3.org/blog/2019/05/w3c-and-whatwg-to-work-together-to-advance-the-open-web-platform/), we have transferred a few specs from the WebPlatform WG to the newly created [WebApps WG](https://www.w3.org/2019/webapps/),  the Input Events spec (same as the Editing API specs) is one of them. It'd be great if you can submit a [new Invited Expert application](https://www.w3.org/ieapp/new) to the Web Applications WG so that we can have your IPR commitment record for this group.

Sorry for the Inconvenience.

**Comment by johanneswilm** [comment]
Hey @siusin . Ok, I filled out the application. Does that mean that the Input events spec is part of the Web Apps WG, while the other editing taskforce specs are part of the WebPlatform WG still?

**Comment by siusin** [comment]
@johanneswilm No, the Editing specs are all in the WebApps WG now. If there are other new proposals from the Editing Task Force in the future, maybe they will be incubated in the WICG.

(Could you [join the WebApps WG](https://www.w3.org/2004/01/pp-impl/114929/join) so that I can revalidate this pull request with the IPR Checker? Thanks! )

File: index.html [hunk 1]
```diff
@@ -1543,6 +1538,10 @@ <h5>
                 <td>
                   A prepopulated {{DataTransfer}} object so that:
                   <ol>
+                    <li>The {{DataTransfer}} object's <a data-cite=
+                    "html/dnd.html#drag-data-store">drag data store</a> is in
```
Comment (marcoscaceres) [inline index.html:1542]: I wonder why this isn't exported by the HTML spec... we should probably check. To export it on the HTML side, we just need to add `data-export=""` to where this is defined. Similarly with the "read-only" definition below.

**Review by marcoscaceres — APPROVED** [review]
This is seems ok, but we should avoid explicit `data-cite=` for [[HTML]] things if we can. Those should be picked up automagically.

**Pull request merged**